### PR TITLE
nixos/saleae-logic: init

### PR DESCRIPTION
**Pull request by chivay**
Add a module supporting Saleae Logic device family.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/hardware/saleae-logic.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.hardware.saleae-logic;
+in
+{
+  options.hardware.saleae-logic = {
+    enable = lib.mkOption {
```
Comment (flokli) [inline nixos/modules/hardware/saleae-logic.nix:8]: mkEnableOption

File: nixos/modules/hardware/saleae-logic.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        Enables udev rules for Saleae Logic devices.
+      '';
+    };
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.saleae-logic-2;
+      defaultText = lib.literalExpression "pkgs.saleae-logic-2";
+      description = ''
+        Saleae Logic package to use.
```
Comment (flokli) [inline nixos/modules/hardware/saleae-logic.nix:21]: This should probably mention you can also add version 1 here?

Right now, I'm using a cheap logic analyzer that only works with version 1, but I could assume I'd buy a new one (and then would want to use version 2).

We could probably even make this a list, so both can be added?

File: nixos/modules/hardware/saleae-logic.nix (hunk 1 above)
Comment (chivay) [reply nixos/modules/hardware/saleae-logic.nix]: Fixed

File: nixos/modules/hardware/saleae-logic.nix (hunk 2 above)
Comment (chivay) [reply nixos/modules/hardware/saleae-logic.nix:15]: To be honest, it doesn't really matter. Both packages contain the same udev rules, so if your analyzer only supports Logic 1.x it should still work file.

Comment (flokli) [reply nixos/modules/hardware/saleae-logic.nix:15]: Indeed!

**Comment by chivay** [comment]
I think it looks fine now. Thanks for the review @flokli !

**Comment by flokli** [comment]
Please squash the different commits once more, to match `CONTRIBUTING.md`.

**Comment by chivay** [comment]
Squashed and rebased onto latest master

**Pull request merged**